### PR TITLE
Simplify input parameters

### LINKED ISSUE #19: Create simplified `liftover_direction` parameter
> **note**: Based on discussion in #13 and other discussions, I think we may want to go with something like a single parameter to determine both (ie. `liftover_direction = "" // Choices: GRCh37ToGRCh38, GRCh38ToGRCh37`); this should simplify the number of parameters needed from the user but still keep the granularity to potentially allow for future additions of assemblies should the need arise.
>
> Though even with this, the source and destination reference fastas would still have to be properly provided by the user as those paths can't be independently generated and inferring assembly from the fasta isn't something we probably want to try to get into

_Originally posted by @yashpatel6 in https://github.com/uclahs-cds/pipeline-StableLift/pull/15#discussion_r1731949839_

### PR DESCRIPTION
**Pull request by nwiltsie**
# Description

Closes #19.

This simplifies the parameter interface by hiding the following parameters from the user:

https://github.com/uclahs-cds/pipeline-StableLift/blob/70cde6086c93f5de34f5c150c5211a6caddf42aa/config/methods.config#L43-L53

Instead, the user is expected to supply:

* `liftover_direction` - must be one of `GRCh37ToGRCh38` or `GRCh38ToGRCh37`
* `fasta_ref_37`
* `fasta_ref_38`

In addition, the branch-specific parameters (`funcotator_data_sources` and `repeat_bed` for SNV, `header_contigs` and `gnomad_rds` for SV) are now conditionally required and validated based on the `variant_caller` parameter.

---

I started three parallel NFTest jobs to test, but unfortunately two of the log file names collided... weirdly, the `SV` and `SNV` jobs won in the `/hot/software/pipeline/.../log-nftest-*.log` files, but the `SV` and `SNV-backward` jobs won in my local `sbatch-nftest-*.log` files!

```console
$ head /hot/software/pipeline/pipeline-StableLift/Nextflow/development/unreleased/nwiltsie-simplify-params/*.log -n 2
==> /hot/software/pipeline/pipeline-StableLift/Nextflow/development/unreleased/nwiltsie-simplify-params/log-nftest-20240830T235010Z.log <==
2024-08-30 23:50:10,433 - NFTest - MainThread - INFO - Beginning tests...
2024-08-30 23:50:10,433 - NFTest - MainThread - INFO - SNV: None

==> /hot/software/pipeline/pipeline-StableLift/Nextflow/development/unreleased/nwiltsie-simplify-params/log-nftest-20240830T235139Z.log <==
2024-08-30 23:51:39,981 - NFTest - MainThread - INFO - Beginning tests...
2024-08-30 23:51:39,981 - NFTest - MainThread - INFO - SV: None
```

```console
$ head sbatch-nftest-20240830T234652Z.log sbatch-nftest-20240830T234651Z.log -n 6
==> sbatch-nftest-20240830T234652Z.log <==
LOG: .env not found in /hot/code/nwiltsie/pipelines/pipeline-StableLift.
LOG: .env not found in /shared/home/nwiltsie.
WARN: unable to find .env. Default values will be used.
================================ NFTEST STARTS =================================
2024-08-30 23:51:39,981 - NFTest - INFO - Beginning tests...
2024-08-30 23:51:39,981 - NFTest - INFO - SV: None

==> sbatch-nftest-20240830T234651Z.log <==
LOG: .env not found in /hot/code/nwiltsie/pipelines/pipeline-StableLift.
LOG: .env not found in /shared/home/nwiltsie.
WARN: unable to find .env. Default values will be used.
================================ NFTEST STARTS =================================
2024-08-30 23:51:39,986 - NFTest - INFO - Beginning tests...
2024-08-30 23:51:39,986 - NFTest - INFO - SNV-backward: None
```

That said, all the jobs are currently running.

# Checklist

- [x] I have read the [code review guidelines](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3187646/Code+Review+Guidelines) and the [code review best practice on GitHub check-list](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List).

- [x] I have reviewed the [Nextflow pipeline standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3193890/Nextflow+pipeline+standardization).

- [x] The name of the branch is meaningful and well formatted following the [standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List), using \[AD_username (or 5 letters of AD if AD is too long)]-\[brief_description_of_branch].

- [x] I have set up or verified the branch protection rule following the [github standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3190380/GitHub+Standards#GitHubStandards-Branchprotectionrule) before opening this pull request.

- [x] I have added my name to the contributors listings in the ``manifest`` block in the `nextflow.config` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

- [x] I have added the changes included in this pull request to the `CHANGELOG.md` under the next release version or unreleased, and updated the date.

- [ ] I have updated the version number in the `metadata.yaml` and `manifest` block of the `nextflow.config` file following [semver](https://semver.org/), or the version number has already been updated. (*Leave it unchecked if you are unsure about new version number and discuss it with the infrastructure team in this PR.*)

- [x] I have tested the pipeline on at least one A-mini sample.

### CODE REVIEW

**Comment by nwiltsie** [comment]
_Beep boop, a script is posting on @nwiltsie's behalf._

The following NFTest cases completed successfully:
* `SNV`
* `SNV-backward`
* `SV`

Summary log file: `/hot/software/pipeline/pipeline-StableLift/Nextflow/development/unreleased/nwiltsie-simplify-params/log-nftest-20240903T214204Z.log`

**Comment by nwiltsie** [comment]
_Finally_, confirmation that everything is working correctly. My auto-posting script fell victim to https://github.com/uclahs-cds/tool-NFTest/pull/70, but I did fix the pipeline bug with a8f5504c4e56c568c7636ab9977d36a97b45ac34.

I manually verified that each test case succeeded:

| Test Case | Log File |
| --------- | -------- |
| SV | `/hot/software/pipeline/pipeline-StableLift/Nextflow/development/unreleased/nwiltsie-simplify-params/log-nftest-20240904T170307Z.log`
| SNV | `/hot/software/pipeline/pipeline-StableLift/Nextflow/development/unreleased/nwiltsie-simplify-params/log-nftest-20240904T164136Z-208595.log`
| SNV-backward | `/hot/software/pipeline/pipeline-StableLift/Nextflow/development/unreleased/nwiltsie-simplify-params/log-nftest-20240904T164136Z-208596.log` |

**Comment by nkwang24** [comment]
@yashpatel6

**Comment by yashpatel6** [comment]
I'll look over this today!

File: config/methods.config [hunk 1]
```diff
@@ -37,22 +39,67 @@ methods {
...
         }
 
-        params.liftover_forward = params.funcotator_data.dest_reference_id == "hg38"
+        def liftover_direction  = params.getOrDefault('liftover_direction', null)
+
+        def forward  = "GRCh37ToGRCh38"
+        def backward = "GRCh38ToGRCh37"
+
+        if (liftover_direction in [forward, backward]) {
+            if (liftover_direction == forward) {
```
Comment (yashpatel6) [inline config/methods.config:67]: **question (non-blocking)**: Is the setup here with the checks and condition intended to make it extensible for potential future additions?

**Review by yashpatel6 — APPROVED** [review]
Looks good to me!

File: config/methods.config (hunk 1 above)
Comment (nwiltsie) [reply config/methods.config:67]: Yeah, that's the idea. It's probably a little over-complicated for just the two cases, but it should be easy to update either side (the user-facing parameters or the internal parameters) without too much hassle for the other.

**Pull request merged**